src/ray/util/filesystem.h

```diff
@@ -0,0 +1,70 @@
+#ifndef RAY_UTIL_FILESYSTEM_H
+#define RAY_UTIL_FILESYSTEM_H
+
+#include <string>
+
+namespace ray {
+
+namespace detail {
+
+static const char *c_str(const std::string &s) { return s.c_str(); }
```

Comment:
Can we get rid of this?

Reply:
I can create a string object for every argument if you'd prefer that? It'll be slower than using the string literals directly though.

Reply:
That would be ok, the path manipulation code shouldn't be in a critical loop anyways.